### ISSUE TITLE
core: include local file "export"

### ISSUE THREAD

**Issue by PatrickJS** [BuilderIO/mitosis#421]
We should include `export`s within the same file.

https://mitosis.builder.io/?outputTab=react&code=import+%7B+useState+%7D+from+%22%40builder.io%2Fmitosis%22%3B%0A%0Aexport+const+defaultValues+%3D+%7B%0A++name%3A+%27hello%27%0A%7D%0A%0Aexport+default+function+MyComponent%28props%29+%7B%0A++const+state+%3D+useState%28%7B%0A++++name%3A+%27%27%0A++%7D%29%3B%0A%0A++return+%28%0A++++%3Cdiv%3E%0A++++++%3Cinput%0A++++++++css%3D%7B%7B%0A++++++++++color%3A+%22red%22%2C%0A++++++++%7D%7D%0A++++++++value%3D%7BdefaultValues.name+%7C%7C+name%7D%0A++++++++onChange%3D%7B%28event%29+%3D%3E+state.name+%3D+event.target.value%7D%0A++++++%2F%3E%0A++++++Hello%21+I+can+run+in+React%2C+Vue%2C+Solid%2C+or+Liquid%21%0A++++%3C%2Fdiv%3E%0A++%29%3B%0A%7D

in Angular we can detect if it's being used then add it to the class. This should help with default values

**Comment by PatrickJS** [comment]
@decadef20 do you think you can do this one? We want to collect all exposed `export` variables probably in MitosisComponent.exportRef what do you think @samijaber?

**Comment by steve8708** [comment]
ah there isn't actually, the original intent was to just straight up copy everything above and below the component into the output, so if you wrote

```
whatever()

export default function MyComponent() { ... }

whatever()
```

we would just copy the added code above/below in the output generator

I think I removed this at some point to not cause confusion, as it's important not to do this as in the past it wouldn't work as expected. now tho it actually might be perfectly ok to support (we'd just have to handle those local vars being forwarded to being available in the component, e.g. we have some logic for this right now with imports)


```
const foo = someSetupCode();

export default function MyComponent() { 
 }
```


tl;dr yeah I think we actually can support, now, arbitrary code above and below components and just copy them over to the same spot in the generators

so the above example can easily just be

```
<template></template>
<script>
econst const foo = someSetupCode()
export default {
  name: "my-component",
};
</script>
```

and if that export is used in the template, it'll need to be forwarded in some cases, like

```
<template></template>
<script>
econst const foo = someSetupCode()
export default {
  data: () => ({ foo }), // Now it can be used in the Vue template, or ideally should be in `methods: { foo }` if it's a method, etc
  name: "my-component",
};
</script>
```

**Comment by PatrickJS** [comment]
yeah we can definitely support this in all frameworks and when you import the file can you choose both default and others.
```jsx
import Component, { helperFn } from './component.mitosis.tsx';
```
or
```jsx
import Component from './component.mitosis.tsx';
import { helperFn } from './component.mitosis.tsx';
```

**Comment by samijaber** [comment]
Sweet, sounds good to me! Definitely an easy feature to add

**Comment by PatrickJS** [comment]
@decadef20 do you want to add this feature? it should be similar to https://github.com/BuilderIO/mitosis/pull/420

**Comment by PatrickJS** [comment]
fixed via https://github.com/BuilderIO/mitosis/pull/453 https://github.com/BuilderIO/mitosis/pull/449
